keras_cv/src/models/segmentation/yolo_v8_segmentation/yolo_v8_segmentation.py

```diff
@@ -0,0 +1,552 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import warnings
+
+from keras_cv.src import bounding_box
+from keras_cv.src import layers
+from keras_cv.src.api_export import keras_cv_export
+from keras_cv.src.backend import keras
+from keras_cv.src.backend import ops
+from keras_cv.src.losses.ciou_loss import CIoULoss
+from keras_cv.src.models.backbones.backbone_presets import backbone_presets
+from keras_cv.src.models.backbones.backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector import (
+    apply_path_aggregation_fpn,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector import (
+    apply_yolo_v8_head as build_YOLOV8_detection_head,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector import (
+    decode_regression_to_boxes,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector import (
+    dist2bbox,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector import (
+    get_anchors,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_detector_presets import (  # noqa: E501
+    yolo_v8_detector_presets,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_label_encoder import (  # noqa: E501
+    YOLOV8LabelEncoder,
+)
+from keras_cv.src.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_conv_bn,
+)
+from keras_cv.src.models.task import Task
+from keras_cv.src.utils.python_utils import classproperty
+from keras_cv.src.utils.train import get_feature_extractor
+
+
+def build_mask_prototypes(x, dimension, num_prototypes, name="prototypes"):
+    """Builds protonet module. The outputs of this tensor are linearly combined
```

Comment:
Usually we have a one line title for the doc string followed by a paragraph of description.